### DISCUSSION TITLE
what is the meaning of publisher in DDS layer?

### DISCUSSION THREAD

**Discussion by wddpfx** [eProsima/Fast-DDS#4249] [General]
In DDS layer, there is participant、topic、publisher、subscriber。Publisher is just a container of dataWriter，But participant is a container of all entity。And participant and dataWriter have a map to rtpsParticipant and rtpsWriter at rpts layer. So what the meaning of publisher or subscriber?

**Comment by JesusPoderoso** [comment]
Hi @wddpfx, thanks for using Fast DDS.
Leaving apart the RTPS layer, a [Publisher](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/publisher/publisher.html) is a factory of [DataWriters](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/publisher/dataWriter/dataWriter.html). Each DataWriter can only publish on a single [Topic](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/topic/topic.html#dds-layer-topic-topic), with its related data-type, so a Publisher can publish on different Topics at the same time. Also, [Publisher QoS](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/publisher/publisher/publisher.html#dds-layer-publisher-publisherqos) are applied to all their DataWriters. 

It occurs the same in the [Subscriber](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/subscriber/subscriber.html) case (and [DataReaders](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/subscriber/dataReader/dataReader.html)).

In the [Participant](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/domain/domainParticipant/domainParticipant.html) case, the participant's configured transport and several QoS are used by all the Publishers and Subscribers contained in the Participant.

**Comment by wddpfx** [comment]
好的，收到了你的邮件，我会尽快查看。
